src/pipecat/processors/user_idle_processor.py

```diff
@@ -32,28 +33,34 @@ def __init__(
         **kwargs,
     ):
         super().__init__(**kwargs)
-
+        self._running = True
         self._callback = callback
         self._timeout = timeout
-
         self._interrupted = False
-
         self._create_idle_task()
 
     async def process_frame(self, frame: Frame, direction: FrameDirection):
         await super().process_frame(frame, direction)
 
+        # Check for end frames before processing
+        if isinstance(frame, (EndFrame, EndTaskFrame)):
+            self._running = False
+            self._idle_event.set()  # Wake up idle task so it can exit
+            await self.cleanup()
+
         await self.push_frame(frame, direction)
 
-        # We shouldn't call the idle callback if the user or the bot are speaking.
-        if isinstance(frame, UserStartedSpeakingFrame):
-            self._interrupted = True
-            self._idle_event.set()
-        elif isinstance(frame, UserStoppedSpeakingFrame):
-            self._interrupted = False
-            self._idle_event.set()
-        elif isinstance(frame, BotSpeakingFrame):
-            self._idle_event.set()
+        # Only process these frames if we're still running
+        if self._running:
```

Comment:
This shouldn't be needed. We shouldn't have more frames after an `EndFrame` or a `CancelFrame`. If so, we have an issue somewhere else.